frankenstein/ring.go

```diff
@@ -0,0 +1,89 @@
+package frankenstein
+
+// Based on https://raw.githubusercontent.com/stathat/consistent/master/consistent.go
+
+import (
+	"errors"
+	"sort"
+	"strconv"
+	"sync"
+)
+
+type uint64s []uint64
+
+func (x uint64s) Len() int           { return len(x) }
+func (x uint64s) Less(i, j int) bool { return x[i] < x[j] }
+func (x uint64s) Swap(i, j int)      { x[i], x[j] = x[j], x[i] }
+
+// ErrEmptyRing is the error returned when trying to get an element when nothing has been added to hash.
+var ErrEmptyRing = errors.New("empty circle")
+
+// Ring holds the information about the members of the consistent hash circle.
+type Ring struct {
+	sync.RWMutex
```

Comment:
We mostly went from anonymous embedded mutexes to named member ones. IMO it has slightly less potential for confusion.